File: ReadyTasks/FSharp/ProjectEuler/Task_21/Task_21/Program.fs
open System

let getDivSum num = 
    Seq.initInfinite (fun x -> x + 2) |> Seq.takeWhile (fun x -> x * x <= num) |>
    Seq.sumBy (fun x -> if x * x = num then x else if (num % x = 0) then x + num / x else 0) |>
    (fun x -> x + 1)

let isAmicable num =
    num = getDivSum (getDivSum num) && (getDivSum num <> num)

let getResult num =
    [1..num] |> List.filter isAmicable |> List.sum

[<EntryPoint>]
let main argv =
    printfn "%A" (getResult 10000)
    0 // return an integer exit code

File: ReadyTasks/CSharp/ProjectEuler/Task_21/Task_21/Program.cs
using System;

namespace Task_21
{
    class Program
    {
        static int GetDivSum(int numb)
        {
            int result = 1;
            for (int i = 2; i * i <= numb; i++)
            {
                if (i * i == numb)
                {
                    result += i;
                }
                else if (numb % i == 0)
                {
                    result += i + numb / i;
                }
            }
            return result;
        }

        static bool IsAmicable(int numb)
        {
            int divSum = GetDivSum(numb);
            return numb != divSum && numb == GetDivSum(divSum);
        }
        static int GetResult(int max)
        {
            int result = 0;
            for (int i = 1; i <= max; i++)
            {
                if (IsAmicable(i))
                {
                    result += i;
                }
            }
            return result;
        }
        static void Main(string[] args)
        {
            Console.WriteLine(GetResult(10000));
        }
    }
}
